cedar-wasm/src/policies_and_templates.rs

```diff
@@ -0,0 +1,250 @@
+use std::str::FromStr;
+
+use cedar_policy::{Policy, PolicySet};
+use serde::{Deserialize, Serialize};
+
+use tsify::Tsify;
+use wasm_bindgen::prelude::*;
+
+#[derive(Tsify, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[tsify(into_wasm_abi, from_wasm_abi)]
+pub enum JsonToPolicyResult {
+    Success { policy_text: String },
+    Error { errors: Vec<String> },
+}
+
+#[wasm_bindgen(js_name = "policyTextFromJson")]
+pub fn policy_text_from_json(json_str: &str) -> JsonToPolicyResult {
+    let parsed_json = match serde_json::from_str(json_str) {
+        Ok(v) => v,
+        Err(e) => {
+            return JsonToPolicyResult::Error {
+                errors: vec![e.to_string()],
+            }
+        }
+    };
+    let policy = Policy::from_json(None, parsed_json);
+    match policy {
+        Ok(p) => JsonToPolicyResult::Success {
+            policy_text: p.to_string(),
+        },
+        Err(e) => JsonToPolicyResult::Error {
+            errors: vec![e.to_string()],
+        },
+    }
+}
+
+#[derive(Tsify, Debug, Serialize, Deserialize, Clone)]
+#[serde(untagged)]
+#[tsify(into_wasm_abi, from_wasm_abi)]
+pub enum PolicyToJsonResult {
+    Success {
+        policy: cedar_policy_core::est::Policy,
+    },
+    Error {
+        errors: Vec<String>,
+    },
+}
+
+#[wasm_bindgen(js_name = "policyTextToJson")]
+pub fn policy_text_to_json(cedar_str: &str) -> PolicyToJsonResult {
+    let parsed_policy = match Policy::from_str(cedar_str) {
+        Ok(policy) => policy,
+        Err(errs) => {
+            return PolicyToJsonResult::Error {
+                errors: errs.errors_as_strings(),
+            };
+        }
+    };
+    match parsed_policy.to_json_policy() {
+        Ok(json_policy) => PolicyToJsonResult::Success {
+            policy: json_policy,
+        },
+        Err(err) => PolicyToJsonResult::Error {
+            errors: vec![err.to_string()],
+        },
+    }
+}
+
+#[derive(Tsify, Debug, Serialize, Deserialize)]
+#[tsify(into_wasm_abi, from_wasm_abi)]
+/// struct that defines the result for the syntax validation function
+pub enum PolicySetParseResult {
+    /// represents successful syntax validation
+    Success { policies: i32, templates: i32 },
+    /// represents a syntax error and encloses a vector of the errors
+    SyntaxError { errors: Vec<String> },
+}
+
+#[wasm_bindgen(js_name = "parsePolicySet")]
+pub fn parse_policy_set(input_policies_str: &str) -> PolicySetParseResult {
```

Comment:
At least for me, I assumed a function named `parse_policy_set` would return policies, but we're returning counts here?

Reply:
I also found this name confusing. I think `check_parse_policy` or something similar would make more sense.